### PR TITLE
profile completion calculation

### PR DESCRIPTION
**Pull request by OnaMosimege**

### CODE REVIEW

File: app/app/views.py [hunk 1]
```diff
@@ -24,9 +24,34 @@ def institutions(request):
...
+
+        """Rating returns % number of empty fields.
+
+        Profile completion weighting is % number of completed fields in 5 star rating.
+
+        Profile completion is calculated using the number of present model fields,
+        model fields have to be added to the institution_dict"""
+
+        rating = (empty_value_count_cal / len(institution_dict)) * 100
+        institution_dict["empty_fields"] = round(rating)
```
Comment (friedelwolff) [inline app/app/views.py:48]: From here on, your code calls the rating "empty_fields", which is a bit inconsistent. I'd say, let's hide the "empty fields" implementation detail here, and only call it "rating" in the context and template. In future the rating might be calculated a different way, and not (only) be about empty fields.

But then, it is a bit odd that a full (good) profile gets a 0 rating, which would usually convey a bad rating. Can I suggest that a rating of 100 is a "good" profile? If it helps in the implementation, you can rather count provided fields, i.e. `completed_fields = sum(1 for value in institution_dict.values() if value)`, but see what works well for you.

File: app/general/tests/test_views_institution.py [hunk 2]
```diff
@@ -31,10 +39,12 @@ def test_institutions_view_correct_projects_returned(self):
...
         institutions = response.context["institutions"]
+
+        self.assertEqual(response.status_code, 200)
         self.assertEqual(len(institutions), 2)
-        self.assertTrue(all(hasattr(inst, "project_count") for inst in institutions))
-        self.assertEqual(institutions[0].project_count, 0)
-        self.assertEqual(institutions[1].project_count, 2)
+        self.assertTrue(all("project_count" in inst for inst in institutions))
+        self.assertEqual(institutions[0]["project_count"], 0)
+        self.assertEqual(institutions[1]["project_count"], 2)
```
Comment (friedelwolff) [inline app/general/tests/test_views_institution.py:47]: Please verify here the rating is correct.

**Review by friedelwolff — CHANGES_REQUESTED** [review]

File: app/general/tests/test_views_institution.py (hunk 2 above)
Comment (OnaMosimege) [reply app/general/tests/test_views_institution.py:47]: I'll add a separate test for rating

File: app/app/views.py (hunk 1 above)
Comment (OnaMosimege) [reply app/app/views.py:48]: Okay, I will change and do it the other way around

**Review by friedelwolff — APPROVED** [review]

**Pull request merged**